src/main/cpp/src/config/json_config.cc

```diff
@@ -35,13 +35,11 @@ const char* g_json_indent_unit = "    ";
 
 rapidjson::Document parse_json_file(const std::string& filename) {
   VERIFY_OR_THROW(filename.length() && "vid/callset mapping file unspecified");
-  char *json_buffer;
+  char *json_buffer = 0;
   size_t json_buffer_length;
   if (TileDBUtils::read_entire_file(filename, (void **)&json_buffer, &json_buffer_length) != TILEDB_OK || !json_buffer || json_buffer_length == 0) {
-    if (json_buffer) {
-      free(json_buffer);
-      throw GenomicsDBConfigException((std::string("Could not open vid/callset mapping file \"")+filename+"\"").c_str());
-    }
```

Comment:
Is it possible for read_entire_file to return error and not have allocated memory for json_buffer?

Reply:
Yes, it is possible. If json_buffer is NULL, then free(buffer) is a NOP. 
From `man 3 free`

```
  The free() function deallocates the memory allocation pointed to by ptr. If ptr is a NULL pointer, no
     operation is performed.
```

Reply:
Learned something new!